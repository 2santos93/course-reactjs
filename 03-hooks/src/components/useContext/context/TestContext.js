import { createContext } from "react";

export const TestContext = createContext({msg: 'no tengo provider :('});