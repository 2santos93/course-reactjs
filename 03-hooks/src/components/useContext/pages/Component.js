export const Component = () => {
  console.log("ENTRE");
  return (
    <div>
      Esto es un componente
    </div>
  )
}
