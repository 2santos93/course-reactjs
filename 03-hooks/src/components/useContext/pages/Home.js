import React, { useContext } from "react";

import { UserContext } from "../context/UserContext";

export const Home = () => {
  const { user, setUser } = useContext(UserContext);
  return (
    <>
      <h1>Home</h1>
      <hr />

      {
        JSON.stringify(user, null, 3)
      }

      <button className="btn btn-primary"
        onClick={() => setUser({id:'1234', name:'nelson'})}
      >Set User</button>
    </>
  );
};
