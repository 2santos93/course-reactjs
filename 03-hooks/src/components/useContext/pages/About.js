import React, { useContext } from "react";
import { 
  useRouteMatch, 
  useParams,
  Switch,
  Route,
  NavLink
} from "react-router-dom";
import { TestContext } from "../context/TestContext";
import { UserContext } from "../context/UserContext";
import { Component } from "./Component";


const SpecificComponent = () => {
  let {id} = useParams();
  console.log("ENTRE");

  return (
    <div>
      Esto es un componente especifico {id}
    </div>
  )
}

export const About = () => {
  const { setUser } = useContext(UserContext);
  const msg = useContext(TestContext);
  console.log(JSON.stringify(msg));

  let match = useRouteMatch();
  console.log(match);

  let params = useParams();
  console.log(params);

  return (
    <div>
      <h1>About</h1>
      <hr />

      <button className="btn btn-danger" onClick={() => setUser({})}>
        Logout
      </button>

      <ul>
        <li>
          <NavLink to={`${match.url}/components`}>Components</NavLink>
        </li>
        <li>
          <NavLink to={`${match.url}/component/1`}>Specific Component</NavLink>
        </li>
      </ul>

      <Switch>
        <Route exact path={`${match.path}/components`}>
            <Component />
        </Route>

        <Route exact path={`${match.path}/component/:id`} component={SpecificComponent} />
      </Switch>
    </div>
  );
};
