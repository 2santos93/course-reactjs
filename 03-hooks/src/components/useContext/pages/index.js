import { Home } from "./Home"
import { Login } from "./Login"
import { About } from "./About"

export {
    Home,
    Login,
    About
}