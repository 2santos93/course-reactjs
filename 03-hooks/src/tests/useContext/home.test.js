import { UserContext } from '../../components/useContext/context/UserContext';
import { Home } from '../../components/useContext/pages/Home';
import {mount} from 'enzyme';
import {act} from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Home page test', () => {
  
  test('should render ok', () => {

    const user = {
      name: 'nelson',
      email: 'test@test.com'
    }

    const wrapper = mount(
      <UserContext.Provider value={{user}} >
          <Home />
      </UserContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();

  })

  test('should set user', () => {

    const setUser = jest.fn();

    const wrapper = mount(
      <UserContext.Provider value={{setUser}} >
          <Home />
      </UserContext.Provider>
    );

    act(() => {
      wrapper.find('button').simulate('click');
    })

    expect(setUser).toHaveBeenCalledWith({id:'1234', name:'nelson'})
  })
    

})
