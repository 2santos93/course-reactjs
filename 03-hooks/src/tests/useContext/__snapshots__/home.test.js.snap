// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test should render ok 1`] = `
Array [
  <h1>
    Home
  </h1>,
  <hr />,
  "{
   \\"name\\": \\"nelson\\",
   \\"email\\": \\"test@test.com\\"
}",
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Set User
  </button>,
]
`;
